node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -44,24 +78,122 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
 
   implicit def executionContext: ExecutionContext = system.dispatcher
 
-  val peers: Vector[Peer]
+  val peersToCheckStack: mutable.Stack[Peer] = mutable.Stack.empty[Peer]
+
+  private def getPeersFromDnsSeeds: Vector[Peer] = {
+    val dnsSeeds = nodeAppConfig.network.dnsSeeds
+    val addresses = dnsSeeds
+      .flatMap(seed => {
+        try {
+          InetAddress
+            .getAllByName(seed)
+        } catch {
+          case _: UnknownHostException =>
+            logger.debug(s"DNS seed $seed is unavailable")
+            Vector()
+        }
+      })
+      .distinct
+      .filter(_.isReachable(500))
+      .map(_.getHostAddress)
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers.toVector
+  }
 
-  private val _peerServices: mutable.Map[Peer, ServiceIdentifier] =
-    mutable.Map.empty
+  private def getPeersFromDb: Future[Vector[Peer]] = {
+    val addressesF: Future[Vector[PeerDB]] =
+      PeerDAO().findAll()
+    val peersF = addressesF.map { addresses =>
+      val inetSockets = addresses.map(a => {
+        NetworkUtil.parseInetSocketAddress(a.address, a.port)
+      })
+      val peers =
+        inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+      peers
+    }
+    peersF
+  }
+
+  private def getPeersFromConf: Vector[Peer] = {
+    val addresses = nodeAppConfig.peers
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers
+  }
+
+  system.scheduler.scheduleWithFixedDelay(initialDelay =
+                                            Duration(10, TimeUnit.SECONDS),
+                                          delay =
+                                            Duration(5, TimeUnit.SECONDS)) {
+    new Runnable() {
+      override def run(): Unit = {
+        if (peersToCheckStack.size < 10)
+          peersToCheckStack.pushAll(getPeersFromDnsSeeds)
+        val peers = for { _ <- 0 to 9 } yield peersToCheckStack.pop()
+        peers.foreach(peer => {
+          addPeer(peer, keepConnection = false)
+          initializePeer(peer)
+        })
```

Comment:
should make this runnable it's own object

Reply:
cc @Christewart should this be added to the node app config?

Reply:
I suppose you mean making it an object as in `object PeerScheduler extends Runnable` and then I would not be able to access members from node as objects don't take any arguments. Making a class would work but then it won't ensure it to be a singleton which is why we are making it an object in the first place (?). How would I go about it?